fabric-object-builders-v0/src/main/java/net/fabricmc/fabric/api/village/v1/VillagerProfessionBuilder.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.village.v1;
+
+import java.util.Set;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.Blocks;
+import net.minecraft.item.Item;
+import net.minecraft.sound.SoundEvent;
+import net.minecraft.util.Identifier;
+import net.minecraft.village.TradeOffers;
+import net.minecraft.village.VillagerProfession;
+import net.minecraft.world.poi.PointOfInterestType;
+
+import net.fabricmc.fabric.impl.object.builder.VillagerProfessionBuilderImpl;
+
+/**
+ * Allows the creation of new {@link VillagerProfession}s.
+ *
+ * <p>Note this does not register any trades to these villagers. To register trades, add a new entry with your profession as the key to {@link TradeOffers#PROFESSION_TO_LEVELED_TRADE}.
+ */
+public interface VillagerProfessionBuilder {
+	/**
+	 * Creates a builder to make a {@link VillagerProfession}.
+	 * @return A new builder.
+	 */
+	static VillagerProfessionBuilder builder() {
```

Comment:
Should `Identifier` be passed in here? and possibly `PointOfInterestType`? Im not too sure, it shouldnt matter too much as you do check when building.

Reply:
I mean I could but I've already specified on the comments for the build method the two absolute requirements.

Reply:
If they are mandatory arguments it seems like it would be sensible to make them required for creating the builder to begin with. We should check how this is handled elsewhere in Fabric API.